run_tests.py

```
@@ -0,0 +1,86 @@
+        print(f"Found {len(test_functions)} test functions")
+        
+        passed = 0
+        failed = 0
+        
+        for test_func in test_functions:
+            try:
+                print(f"  Running {test_func.__name__}...")
+                test_func()
+                print(f"  ✅ {test_func.__name__} passed")
+                passed += 1
+            except Exception as e:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)